pkg/iac-providers/docker/v1/load-dir.go

```diff
@@ -0,0 +1,107 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package dockerv1
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/accurics/terrascan/pkg/iac-providers/output"
+	"github.com/accurics/terrascan/pkg/results"
+	"github.com/accurics/terrascan/pkg/utils"
+	"github.com/hashicorp/go-multierror"
+	"go.uber.org/zap"
+)
+
+const (
+	dockerDirectory        string = "docker"
+	resourceTypeDockerfile string = "dockerfile"
+)
+
+// LoadIacDir loads the docker file specified in given folder.
+func (dc *DockerV1) LoadIacDir(absRootDir string, nonRecursive bool) (output.AllResourceConfigs, error) {
+	allResourcesConfig := make(map[string][]output.ResourceConfig)
+	fileMap, err := utils.FindFilesBySuffix(absRootDir, []string{DockerFileName})
+	if err != nil {
+		zap.S().Errorf("error while searching for iac files", zap.String("root dir", absRootDir), zap.Error(err))
+		return allResourcesConfig, multierror.Append(dc.errIacLoadDirs, results.DirScanErr{IacType: "docker", Directory: absRootDir, ErrMessage: err.Error()})
+	}
+
+	for fileDir, files := range fileMap {
```

Comment:
Can you try re-writing this using DRY principle? like, re-using LoadIacFile function here? You can get a reference from the k8s iac-provider implementation.